src/main/scala/org/apache/spark/deploy/armada/submit/ArmadaClientApplication.scala

```diff
@@ -650,8 +622,12 @@ private[spark] class ArmadaClientApplication extends SparkApplication {
   ): String = {
     val driverResponse = armadaClient.submitJobs(queue, jobSetId, Seq(driver))
     val driverJobId    = driverResponse.jobResponseItems.head.jobId
+    val error =
+      if (driverResponse.jobResponseItems.head.error.nonEmpty)
+        driverResponse.jobResponseItems.head.error
+      else "none"
```

Comment:
```suggestion
    val error = driverResponse.jobResponseItems.head.error.getOrElse("none")
```

Reply:
Hey @EnricoMi  It looks like this is an invalid change, and I need to back it out.  the "error" field is a String, and not an Option so "getOrElse()" is not available.

Reply:
Then we could do
```suggestion
    val error = Some(driverResponse.jobResponseItems.head.error)
      .filter(_.nonEmpty).getOrElse("none")
```

Reply:
Thanks @EnricoMi I accepted your fix and also fixed the executor version the same way:
 https://github.com/armadaproject/armada-spark/blob/63ca679f06af11f246da766f539e31c542337120/src/main/scala/org/apache/spark/deploy/armada/submit/ArmadaClientApplication.scala#L641